pkg/apis/sessions/session_state.go

```diff
@@ -82,6 +87,15 @@ func (s *SessionState) EncodeSessionState(c encryption.Cipher) (string, error) {
 				return "", err
 			}
 		}
+
+		ss.Groups = append([]string(nil), ss.Groups...)
```

Comment:
I think with the current session encryption architecture, this is the best we can do.  It will just cause ALOT of bloat (each encrypted string gets 16 bytes added as the AES initialization vector + 33% for base64 encoding).

This PR that cleans that issue up is probably 1-2 weeks out if you just want to wait: https://github.com/oauth2-proxy/oauth2-proxy/pull/632